test/currencies/currencies.api-test.ts

```diff
@@ -0,0 +1,169 @@
+    it('should return 200 on GET `/currencies/exchange?source=GBP&target=AED&exchangeRateDate=2021-01-26`', async () => {
+      const { status, body } = await api.get('/currencies/exchange?source=GBP&target=AED&exchangeRateDate=2021-01-26');
+      expect(status).toBe(200);
+      expect(body).toEqual(expectedResult);
+    });
+
+    it('should return 200 on GET `/currencies/exchange?source=GBP&target=AED`', async () => {
+      const { status, body } = await api.get('/currencies/exchange?source=GBP&target=AED');
+      expect(status).toBe(200);
+      expect(body).toEqual(expectedResult);
+    });
+
+    it('should return 200 on GET `currencies/exchange?source=USD&target=GBP`', async () => {
+      const { status, body } = await api.get('/currencies/exchange?source=USD&target=GBP');
+      expect(status).toBe(200);
+      expect(body).toEqual(expectedResult);
+    });
+
+    it('should return 404 on GET `/currencies/exchange?source=AED&target=GBP`', async () => {
+      const { status, body } = await api.get('/currencies/exchange?source=AED&target=GBP');
+      expect(status).toBe(404);
+      expect(body).toEqual({
+        statusCode: 404,
+        message: 'The selected exchange rate is not available',
+        error: 'Not Found',
+      });
+    });
+
+    it('should return 400 on GET `/currencies/exchange?source=GBP&target=abc`', async () => {
+      const { status, body } = await api.get('/currencies/exchange?source=GBP&target=abc');
+      expect(status).toBe(400);
+      expect(body).toEqual({
+        statusCode: 400,
+        message: ['target must be a valid ISO4217 currency code'],
+        error: 'Bad Request',
+      });
+    });
+
+    it('should return 400 on GET `/currencies/exchange?source=abc&target=AED`', async () => {
+      const { status, body } = await api.get('/currencies/exchange?source=abc&target=AED');
+      expect(status).toBe(400);
+      expect(body).toEqual({
+        statusCode: 400,
+        message: ['source must be a valid ISO4217 currency code'],
+        error: 'Bad Request',
+      });
+    });
+
+    it('should return 400 on GET `/currencies/exchange?source=abc&target=def`', async () => {
+      const { status, body } = await api.get('/currencies/exchange?source=abc&target=def');
+      expect(status).toBe(400);
+      expect(body).toEqual({
+        statusCode: 400,
+        message: ['source must be a valid ISO4217 currency code', 'target must be a valid ISO4217 currency code'],
+        error: 'Bad Request',
+      });
+    });
+
+    it('should return 400 on GET `/currencies/exchange?source=GBP`', async () => {
+      const { status, body } = await api.get('/currencies/exchange?source=GBP');
+      expect(status).toBe(400);
+      expect(body).toEqual({
+        statusCode: 400,
+        message: ['target must be a valid ISO4217 currency code'],
+        error: 'Bad Request',
+      });
+    });
+
+    it('should return 400 on GET `/currencies/exchange?target=GBP`', async () => {
```

Comment:
Can we please add a test case where both `source` and `target` and empty?

Reply:
I've added this above